src/System.Buffers/tests/BasicUnitTests.cs

```diff
@@ -40,5 +40,57 @@ public void ByteSpanEmptyCreateArrayTest()
             var array = empty.CreateArray();
             Assert.Equal(0, array.Length);
         }
+        
+        [Fact]
+        public unsafe void ByteSpanEqualsTestsTwoDifferentBuffersSameValues()
```

Comment:
Test name is confusing. It was supposed to be: two different instances of buffer with the same values